tests/scripts/thread-cert/v1_2_test_multicast_registration.py

```diff
@@ -0,0 +1,254 @@
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import unittest
+
+import command
+import config
+import mle
+import thread_cert
+
+LEADER_1_2 = 1
+MED_1_2 = 2
+SED_1_2 = 3
+MED_1_1 = 4
+SED_1_1 = 5
+
+WAIT_ATTACH = 5
+WAIT_REDUNDANCE = 3
+
+MA1_LINKLOCAL = 'ff02::1:2:3:4'
+MA2_ADMINSCOPE = 'ff04::1:2:3:4'
+"""
+ Topology
+
+             SED_1_2
+                |
+                |
+MED_1_2 --- LEADER_1_2 --- MED_1_1
+                |
+                |
+             SED_1_1
+
+
+ 1) Bring up Leader_1_2.
+
+ 2) Bring up MED_1_2:
+    a) add MA1_LINKLOCAL which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+ 3) Bring up SED_1_2:
+    a) add MA1_LINKLOCAL which would be registered in AddressRegistrationTLV of Child Update Request
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+ 4) Bring up MED_1_1:
+    a) add MA1_LINKLOCAL which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+    b) add MA2_ADMINSCOPE which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+
+ 5) Bring up SED_1_1:
+    a) add MA1_LINKLOCAL which would be registered in AddressRegistrationTLV of Child Update Request
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+"""
+
+
+class TestMulticastRegistration(thread_cert.TestCase):
+    topology = {
+        LEADER_1_2: {
+            'version': '1.2',
+            'whitelist': [MED_1_2, SED_1_2, MED_1_1, SED_1_1],
+        },
+        MED_1_2: {
+            'mode': 'rsn',
+            'version': '1.2',
+            'whitelist': [LEADER_1_2],
+        },
+        SED_1_2: {
+            'mode': 'sn',
+            'version': '1.2',
+            'whitelist': [LEADER_1_2],
+        },
+        MED_1_1: {
+            'mode': 'rsn',
+            'version': '1.1',
+            'whitelist': [LEADER_1_2],
+        },
+        SED_1_1: {
+            'mode': 'sn',
+            'version': '1.1',
+            'whitelist': [LEADER_1_2],
+        },
+    }
+    """All nodes are created with default configurations"""
+
+    def test(self):
+
+        # 1) Bring up Leader_1_2.
+        self.nodes[LEADER_1_2].start()
+        WAIT_TIME = WAIT_ATTACH
+        self.simulator.go(WAIT_TIME)
```

Comment:
```suggestion
        self.simulator.go(WAIT_ATTACH)
```
Variable `WAIT_TIME` seems redundant. Many same issues follows.

Reply:
Any reason to make `go()` line same?

Reply:
just personal preference, no strong specific reason. if have to say one, `WAIT_TIME = ` is shorter than `self.simulator.go(`, a bit less chance to cause line break.

Reply:
Personal preference too. I am fine to keep the code as it is.